concrete/controllers/single_page/dashboard/users/add.php

```diff
@@ -40,50 +38,21 @@ public function view()
     public function submit()
     {
         $assignment = PermissionKey::getByHandle('edit_user_properties')->getMyAssignment();
-        $vals = Loader::helper('validation/strings');
-        $valt = Loader::helper('validation/token');
-        $valc = Loader::helper('concrete/validation');
 
         $username = trim($_POST['uName']);
-        $username = preg_replace("/\s+/", " ", $username);
+        $username = preg_replace("/\s+/", ' ', $username);
         $_POST['uName'] = $username;
 
         $password = $_POST['uPassword'];
 
-        if (!$vals->email($_POST['uEmail'])) {
-            $this->error->add(t('Invalid email address provided.'));
-        } elseif (!$valc->isUniqueEmail($_POST['uEmail'])) {
-            $this->error->add(t("The email address '%s' is already in use. Please choose another.", $_POST['uEmail']));
-        }
+        $this->app->make('validator/user/name')->isValid($username, $this->error);
 
-        if (strlen($username) < Config::get('concrete.user.username.minimum')) {
-            $this->error->add(t('A username must be at least %s characters long.', Config::get('concrete.user.username.minimum')));
-        }
-
-        if (strlen($username) > Config::get('concrete.user.username.maximum')) {
-            $this->error->add(t('A username cannot be more than %s characters long.', Config::get('concrete.user.username.maximum')));
-        }
-
-        if (strlen($username) >= Config::get('concrete.user.username.minimum') && !$valc->username($username)) {
-            if (Config::get('concrete.user.username.allow_spaces')) {
-                $this->error->add(t('A username may only contain letters, numbers, spaces (not at the beginning/end), dots (not at the beginning/end), underscores (not at the beginning/end).'));
-            } else {
-                $this->error->add(t('A username may only contain letters, numbers, dots (not at the beginning/end), underscores (not at the beginning/end).'));
-            }
-        }
-
-        if (!$valc->isUniqueUsername($username)) {
-            $this->error->add(t("The username '%s' already exists. Please choose another", $username));
-        }
-
-        if ($username == USER_SUPER) {
-            $this->error->add(t('Invalid Username'));
-        }
+        $this->app->make('validator/password')->isValid($password, $this->error);
 
-        \Core::make('validator/password')->isValid($password, $this->error);
+        $this->app->make('validator/user/email')->isValid($_POST['uEmail'], $this->error);
```

Comment:
I don't think we need to (or should) use `$_POST` here, can you update this to use the request object?